backend/src/main/java/com/bakdata/conquery/models/datasets/concepts/MatchingStats.java

```diff
@@ -21,32 +21,41 @@
     @JsonIgnore
     private transient CDateRange span;
 
+    @JsonIgnore
+    private transient long numberOfEvents = -1L;
+
+    @JsonIgnore
+    private transient long numberOfEntities = -1L;
+
     public long countEvents() {
-        return entries.values().stream().mapToLong(Entry::getNumberOfEvents).sum();
+        if (numberOfEvents == -1L) {
+            numberOfEvents = entries.values().stream().mapToLong(Entry::getNumberOfEvents).sum();
+        }
+        return numberOfEvents;
     }
 
 
-    public long countEntities() {
-
-        return entries.values().stream().mapToLong(Entry::getNumberOfEntities).sum();
+    public synchronized long countEntities() {
```

Comment:
Das synchronized sollte weg von der Methode. Die Funktion wird oft auf gerufen, dass jedes mal in einen synchronizierten Context gesprungen wird macht den Aufruf langsam.

Benutze das Pattern, dass ich in dem Kommentar geschrieben habe.